sklearn/utils/multiclass.py

```diff
@@ -252,8 +252,13 @@ def type_of_target(y):
 
     y = np.asarray(y)
 
-    # The old sequence of sequences format
     try:
+        # labels in bytes format
+        if isinstance(y[0], bytes):
+            raise ValueError('Labels represented as bytes is not supported.'
+                             ' Convert the labels to a supported format.'
+                             ' For example, y = y.astype(str)')
```

Comment:
I am not sure that `y=y.astype(str)` will be the right way to solve the problem:

* it expects that we have a NumPy array;
* it will convert into a NumPy string array that we don't always support properly if I am not wrong.